lib/userAgent.js

```diff
@@ -34,6 +34,11 @@ module.exports = function (userAgentString) {
   this.uaOSVersion = getVersion(userAgentData.os) || null
   this.uaDeviceType = getDeviceType(userAgentData) || null
 
+  if (! this.uaBrowser && ! this.uaOS) {
+    // In the worst case, fall back to a truncated user agent string
+    this.uaBrowser = truncate(userAgentString || '')
```

Comment:
@rfk, as mentioned in our 1:1 and at https://github.com/mozilla/fxa-content-server/pull/3898#issuecomment-229878173, this is my proposed fallback code for unparseable user-agent strings.